ArduCopter/baro_ground_effect.cpp

```diff
@@ -67,3 +71,18 @@ void Copter::update_ground_effect_detector(void)
     ahrs.setTakeoffExpected(gndeffect_state.takeoff_expected);
     ahrs.setTouchdownExpected(gndeffect_state.touchdown_expected);
 }
+
+// update ekf terrain height stable setting
+// when set to true, this allows the EKF to stabilize the normally barometer based altitude using a rangefinder
+// this is not related to terrain following
+void Copter::update_ekf_terrain_height_stable()
+{
+    // set to false if no position estimate
+    if (!position_ok() && !optflow_position_ok()) {
```

Comment:
Doesn't this imply you have to have a GPS lock which I think is too restrictive?

Reply:
@andyp1per, yes it does but I don't think we can know if the terrain below the vehicle is stable or not unless we know how quickly the vehicle is moving horizontally.